silica/verilog.py

```diff
@@ -122,34 +184,50 @@ def compile_statements(states, _tab, one_state, width_table, statements):
                         # if these_conds:
                         #     conds.append(" & ".join(these_conds))
             if not one_state:
-                conds = [f"(yield_state == {yield_id})" for yield_id in yields]
+                conds = [module.get_vars()["yield_state"] == yield_id for yield_id in yields]
+                conds_old = [f"(yield_state == {yield_id})" for yield_id in yields] # TODO: remove
             process_statement(statement)
             if conds:
-                cond = " | ".join(conds)
-                verilog_source += f"\n{_tab}if ({cond}) begin"
+                cond = reduce(vg.Lor, conds)
+                seq.If(cond)(
+                    vg.Subst(translate_value(module, statement.targets[0]), translate_value(module, statement.value), 1)
+                )
+                # TODO: remove
+                cond_old = " | ".join(conds_old)
+                verilog_source += f"\n{_tab}if ({cond_old}) begin"
                 verilog_source += f"\n{_tab + offset}" + astor.to_source(statement).rstrip() + ";"
                 verilog_source += f"\n{_tab}end"
             else:
                 verilog_source += f"\n{_tab}" + astor.to_source(statement).rstrip() + ";"
+                seq(
+                    vg.Subst(translate_value(module, statement.targets[0]), translate_value(module, statement.value), 1)
+                )
         else:
             process_statement(statement)
             verilog_source += f"\n{_tab}" + astor.to_source(statement).rstrip() + ";"
+            seq(
+                vg.Subst(translate_value(module, statement.targets[0]), translate_value(module, statement.value), 1)
+            )
     temp_var_source = ""
     return verilog_source, temp_var_source
 
 
-def compile_states(states, one_state, width_table, strategy="by_statement"):
+def compile_states(module, states, one_state, width_table, strategy="by_statement"):
+    seq = vg.TmpSeq(module, module.get_ports()["CLK"])
     always_source = """\
     always @(posedge CLK) begin\
 """
     tab = "    "
     temp_var_source = ""
     if strategy == "by_path":
+        raise NotImplementedError("by_path is not implemented.")
+        # TODO: do this
```

Comment:
We can hold off on this for now, since the SSA backend is the intended one to be used.